### PR TITLE
Ability to create lexical graph only

### PR DESCRIPTION
**Pull request by stellasia**
# Description

This PR adds a `LexicalGraphBuilder` component that enables users to create the lexical graph without entity/relation extraction (see added example). It also updates the `ERExtraction` component to use the new LexicalGraphBuilder. 

### Why

Pipelines that can be created after this PR:

1. Pdf Loader
2. Text splitter
3. Chunk embedder
4. Lexical Graph Builder
5. Neo4j Writer

A follow-up PR (#135) will implement the `Neo4jChunkReader` component that will enable to start the ER extraction from chunks already saved in the DB.

### Configuration

All parameters for the lexical graph are stored in a `LexicalGraphConfig` object (chunk and document node labels, relationship types, property names). This object is totally optional, default values are provided for all parameters.

### Future

We could consider deprecating the `create_lexical_graph` behavior in the `EntityRelationExtractor` and force users who want to build the lexical graph to use the `LexicalGraphBuilder -> Neo4jWriter` pattern.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

Complexity: Medium

## How Has This Been Tested?
- [x] Unit tests
- [x] E2E tests
- [x] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] Unit tests have been updated
- [x] E2E tests have been updated
- [x] Examples have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed
- [x] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: docs/source/user_guide_kg_builder.rst [hunk 1]
```diff
@@ -166,12 +167,41 @@ Example usage:
...
+If OpenAI is not an option, see :ref:`embedders` to learn how to use other supported embedders.
 
 The embeddings are added to each chunk metadata, and will be saved as a Chunk node property in the graph if
 `create_lexical_graph` is enabled in the `EntityRelationExtractor` (keep reading).
 
 
+Lexical Graph Builder
+=====================
+
+Once the chunks are extracted and, if needed, embedded, a graph can be created. The lexical graph contains:
```
Comment (willtai) [inline docs/source/user_guide_kg_builder.rst:179]:
```suggestion
Once the chunks are extracted and embedded (when required), a graph can be created. The lexical graph contains:
```

I think this reads better?

File: src/neo4j_graphrag/experimental/components/entity_relation_extractor.py [hunk 2]
```diff
@@ -414,34 +311,43 @@ async def run(
         self,
         chunks: TextChunks,
         document_info: Optional[DocumentInfo] = None,
+        lexical_graph_config: Optional[LexicalGraphConfig] = None,
         schema: Union[SchemaConfig, None] = None,
         examples: str = "",
         **kwargs: Any,
     ) -> Neo4jGraph:
-        """Perform entity and relation extraction for all chunks in a list."""
+        """Perform entity and relation extraction for all chunks in a list.
```
Comment (willtai) [inline src/neo4j_graphrag/experimental/components/entity_relation_extractor.py:319]: Could we define 'lexical graph' either here or in the user guide?

Comment (stellasia) [reply src/neo4j_graphrag/experimental/components/entity_relation_extractor.py:319]: This is what I tried to do [here](https://github.com/neo4j/neo4j-graphrag-python/pull/127/files#diff-c37850688512e77f403fe5c0f70c11f569a5ae233777f9859877f2f49dfb4619R179-R184). What is missing? I'll add a note in this docstring anyway to clarify the behavior of `create_lexical_graph` and `lexical_graph_config`.

Comment (willtai) [reply src/neo4j_graphrag/experimental/components/entity_relation_extractor.py:319]: Perhaps how is the term 'lexical graph' from any other knowledge graph. I think a newcomer will be confused with that

**Review by willtai — APPROVED** [review]
Nice work!

File: src/neo4j_graphrag/experimental/components/entity_relation_extractor.py (hunk 2 above)
Comment (stellasia) [reply src/neo4j_graphrag/experimental/components/entity_relation_extractor.py:319]: Added more details and link to the user guide, let me know if you think more information is needd.

**Pull request merged**